extensions/amp-jwplayer/0.1/amp-jwplayer.js

```diff
@@ -52,26 +65,48 @@ class AmpJWPlayer extends AMP.BaseElement {
 
   /** @override */
   buildCallback() {
+    const el = this.element;
     this.contentid_ = userAssert(
-        (this.element.getAttribute('data-playlist-id') ||
-      this.element.getAttribute('data-media-id')),
+        (el.getAttribute('data-playlist-id') ||
+        el.getAttribute('data-media-id')),
         'Either the data-media-id or the data-playlist-id ' +
-      'attributes must be specified for <amp-jwplayer> %s',
-        this.element);
+        'attributes must be specified for <amp-jwplayer> %s',
+        el);
 
     this.playerid_ = userAssert(
-        this.element.getAttribute('data-player-id'),
+        el.getAttribute('data-player-id'),
         'The data-player-id attribute is required for <amp-jwplayer> %s',
-        this.element);
+        el);
+
+    this.contentSearch_ = el.getAttribute('data-content-search') ||
+        '';
+    this.contentContextual_ = el.getAttribute('data-content-contextual') ||
+        false;
+    this.contentRecency_ = el.getAttribute('data-content-recency') ||
+        '';
+    this.contentBackfill_ = el.getAttribute('data-content-backfill') ||
+        false;
   }
 
 
   /** @override */
   layoutCallback() {
     const iframe = this.element.ownerDocument.createElement('iframe');
-    const src = 'https://content.jwplatform.com/players/' +
-      encodeURIComponent(this.contentid_) + '-' +
-      encodeURIComponent(this.playerid_) + '.html';
+    const searchVal = this.getContextualVal();
+    const qs = {
+      'search': searchVal || undefined,
+      'contextual': this.contentContextual_ || undefined,
+      'recency': this.contentRecency_ || undefined,
+      'backfill': this.contentBackfill_ || undefined,
+    };
+  
+    const cid = encodeURIComponent(this.contentid_);
+    const pid = encodeURIComponent(this.playerid_);
+
+    const baseUrl = `https://content.jwplatform.com/players/${cid}-${pid}.html`;
+    const src = Object.keys(qs).length ?
```

Comment:
`addParamsToUrl` handles empty param arguments. You don't need this `Object.keys(qs).length ?` check.

Reply:
@robwalch I will remove this. Had meant to. It was a test to see if this logic would resolve the issue from `gulp check-types`